server/conn.go

```diff
@@ -2156,11 +2156,50 @@ func (cc getLastStmtInConn) PProfLabel() string {
 	case mysql.ComStmtReset:
 		return "ResetStmt"
 	case mysql.ComQuery, mysql.ComStmtPrepare:
-		return parser.Normalize(queryStrForLog(string(hack.String(data))))
+		return parser.Normalize(util2.QueryStrForLog(string(hack.String(data))))
 	case mysql.ComStmtExecute, mysql.ComStmtFetch:
 		stmtID := binary.LittleEndian.Uint32(data[0:4])
-		return queryStrForLog(cc.preparedStmt2StringNoArgs(stmtID))
+		return util2.QueryStrForLog(cc.preparedStmt2StringNoArgs(stmtID))
 	default:
 		return ""
 	}
 }
+
+var (
+	pprofLabelOfUseDB      = parser.DigestNormalized("UseDB")
+	pprofLabelOfListFields = parser.DigestNormalized("ListFields")
+	pprofLabelOfCloseStmt  = parser.DigestNormalized("CloseStmt")
+	pprofLabelOfResetStmt  = parser.DigestNormalized("ResetStmt")
+)
+
+// PProfLabelNormalizedAndDigest return sql and sql_digest label used to tag pprof.
+func (cc getLastStmtInConn) pprofLabelNormalizedAndDigest() (string, string) {
+	if len(cc.lastPacket) == 0 {
+		return "", ""
+	}
+	cmd, data := cc.lastPacket[0], cc.lastPacket[1:]
+	switch cmd {
+	case mysql.ComInitDB:
+		return "UseDB", pprofLabelOfUseDB.String()
+	case mysql.ComFieldList:
+		return "ListFields", pprofLabelOfListFields.String()
```

Comment:
They are even not statements. Maybe can be ignored?

Reply:
Ok, reasonable.